src/OrbitGl/TrackManager.cpp

```diff
@@ -58,6 +58,8 @@ std::vector<Track*> TrackManager::GetAllTracks() const {
   for (const auto& track : all_tracks_) {
     tracks.push_back(track.get());
   }
+
+  std::lock_guard<std::recursive_mutex> lock(mutex_);
```

Comment:
Why instead not using the existent method instead of duplicating this mutex?

Reply:
I wanted to avoid moving the second temporary vector (result of `GetFrameTracks`) around. But I agree, using `GetFrameTracks` would be more logical and less error-prone, I think I'll change it, thanks.